.github/workflows/_checkmarx.yml

```diff
@@ -20,8 +20,8 @@ on:
         required: true
 
 concurrency:
-  group: ${{ github.repository }}-${{ github.workflow }}-checkmarx-${{ github.event.pull_request.number || github.head_ref || github.ref }}
-  cancel-in-progress: true
+  group: ${{ github.repository }}-${{ github.workflow }}-checkmarx-${{ github.event.pull_request.number || github.run_id }}
+  cancel-in-progress: ${{ contains(github.event_name, 'pull_request') }}
```

Comment:
Related to the above, we're really just looking at using the concurrency group _for_ cancellations and want this to be true as a result. I don't understand the use case for maintaining rolling runs of these scans on a branch -- we just need the latest one.